Student/Student/Submissions/SubmissionButton/SubmissionButtonPresenter.swift

```diff
@@ -175,9 +177,19 @@ extension SubmissionButtonPresenter: FilePickerControllerDelegate {
 
     func submit(_ controller: FilePickerViewController) {
         guard let assignment = assignment else { return }
-        let context = FileUploadContext.submission(courseID: assignment.courseID, assignmentID: assignment.id, comment: nil)
-        controller.dismiss(animated: true) {
-            UploadManager.shared.upload(batch: self.batchID, to: context)
+
+        if assignment.submissionTypes.contains(.media_recording) {
+            controller.dismiss(animated: true) { [weak self] in
+                guard let file = controller.files?.first, let url = file.localFileURL else { return }
+                self?.submitMediaType(.video, url: url, callback: { error in
+                    UploadManager.shared.complete(file: file, error: error)
+                })
```

Comment:
I think you're on the right track. I was thinking we would want to use the UploadManager to upload the files (instead of using `submitMediaType` and `UploadMedia`) so that it can all be done in the background but I don't know if that's really necessary. This is definitely a lot simpler.

I think calling `complete` is a good idea. You don't want to capture `file` in that block though since it's a NSManagedObject and is not thread safe. You should probably look it up by its `objectID` instead.

```swift
let objectID = file.objectID
self?.submitMediaType(.video, url: url, callback: { error in
    guard let file = UploadManager.shared.viewContext.existingObject(with: objectID) as? File else { return }
    UploadManager.shared.complete(file: file, error: error)
}
```

Maybe you don't really have to call complete though. The important thing that it does is it deletes the local file so it's not hogging up user disk space. If you do just that in the completion handler then maybe that's enough?